features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/SystemRequirementConfiguration.kt

```diff
@@ -0,0 +1,113 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay
+
+import com.datadog.android.api.InternalLogger
+import com.datadog.android.sessionreplay.internal.prerequisite.CpuRequirementChecker
+import com.datadog.android.sessionreplay.internal.prerequisite.DefaultSystemRequirement
+import com.datadog.android.sessionreplay.internal.prerequisite.MemoryRequirementChecker
+import com.datadog.android.sessionreplay.internal.prerequisite.SystemRequirement
+
+/**
+ * This class defines the system requirements necessary to enable the Session Replay feature.
+ *
+ * @param minCPUCoreNumber The minimum CPU core number.
+ * @param minRAMSize The minimum RAM in megabytes.
+ */
+class SystemRequirementConfiguration internal constructor(
+    val minCPUCoreNumber: Int,
+    val minRAMSize: Int,
+    private var systemRequirement: SystemRequirement? = null
+) {
+
+    internal fun runIfMeetRequirements(
+        internalLogger: InternalLogger,
+        runnable: () -> Unit
+    ) {
+        if (systemRequirement == null) {
+            systemRequirement = DefaultSystemRequirement()
+        }
```

Comment:
```suggestion
    private var systemRequirement: SystemRequirement = DefaultSystemRequirement()
) {

    internal fun runIfMeetRequirements(
        internalLogger: InternalLogger,
        runnable: () -> Unit
    ) {
```

Reply:
Here I leave the constructor as `null` by default because I don't want `DefaultSystemRequirement` to be initialised directly because it uses `Handler(Looper.getMainLooper())` as default handler, and I have `NONE` and `DEFAULT` instance in `companion object` will use it anyway, which can cause trouble in unit test.

so I prefer assign it when really use it. what do you think?

Reply:
I'm not sure what is the problem with unit-test, can you elaborate?

Reply:
yes, so we have two companion objects `DEFAULT`, `NONE` in `SystemRequirementConfiguration` class, they will be initialised directly at the very beginning of the runtime.
they call the constructor to build an instance of `SystemRequirementConfiguration`, so if we have `DefaultSystemRequirement` in the constructor, they will initialise it also with ` Handler(Looper.getMainLooper())` as the default handler.

In unit test, we need some setup to make `Looper.getMainLooper` become non-null, and if I understand correctly,  the setup happens during `@BeforeEach`, which will be too late.  And unit test throws error due to the null result of `Looper.getMainLooper`